homeassistant/components/sma/__init__.py

```diff
@@ -134,3 +138,13 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         data[PYSMA_REMOVE_LISTENER]()
 
     return unload_ok
+
+
+async def update_listener(hass, entry):
+    """Handle options update."""
+    _LOGGER.info("Change SMA language to: %s", entry.options.get(CONF_LANGUAGE))
+    ok_result = await async_unload_entry(hass, entry)
+    if ok_result:
+        ok_result = await async_setup_entry(hass, entry)
+
+    return ok_result
```

Comment:
Splitting up the results could improve the readability here:

```python
    ok_result = False
    unloaded = await async_unload_entry(hass, entry)
    if unloaded:
        ok_result = await async_setup_entry(hass, entry)

    return ok_result
```

Reply:
Thanks for your input, I changed the code according your suggestion.